src/unlimited.cpp

```diff
@@ -289,35 +289,35 @@ void HandleExpeditedBlock(CDataStream& vRecv,CNode* pfrom)
     {
       CXThinBlock thinBlock;
       vRecv >> thinBlock;
+      uint256 blkHash = thinBlock.header.GetHash();
+      CInv inv(MSG_BLOCK, blkHash);
 
-      CInv inv(MSG_BLOCK, thinBlock.header.GetHash());
-
-      BlockMap::iterator mapEntry = mapBlockIndex.find(thinBlock.header.GetHash());
+      BlockMap::iterator mapEntry = mapBlockIndex.find(blkHash);
       CBlockIndex *blkidx = NULL;
       unsigned int status = 0;
       if (mapEntry != mapBlockIndex.end())
 	{
 	  blkidx = mapEntry->second;
-	  status = blkidx->nStatus;
+	  if (blkidx) status = blkidx->nStatus;
```

Comment:
I don't think it is expected normal operations to store a nullptr in the map. Unless this is the case, this should at least log, and probably bail because something went horribly wrong.